src/libs/Network.js

```diff
@@ -107,12 +127,17 @@ function processNetworkRequestQueue() {
         if (!networkRequestQueue.length) {
             return;
         }
-
-        // If we have a request then we need to check if it can be persisted in case we close the tab while offline
-        const retryableRequests = _.filter(networkRequestQueue, request => (
-            !request.data.doNotRetry && request.data.persist
-        ));
-        Onyx.set(ONYXKEYS.NETWORK_REQUEST_QUEUE, retryableRequests);
+        const retryableRequests = [];
+
+        // If we have a request then we need to check if it can be persisted in case we close the tab while offline.
+        // We filter persisted requests from the normal Queue to remove duplicates
+        networkRequestQueue = _.reject(networkRequestQueue, (request) => {
```

Comment:
>We filter persisted requests from the normal Queue to remove duplicates

Which duplicates are we talking about here? Can you provide an example? It's hard to visualize what you mean or why we are removing these from the normal request queue in memory.

Wouldn't this mean that if I add a comment while offline that it will be "removed" from the queue and not sent again when I come back online?

Reply:
Can you link me to where this happens?

Reply:
bump

Reply:
When use comes online on line 37 we process the queue by fake merge which I will optimize based on the suggestion.